src/store/recordAction.ts

```diff
@@ -0,0 +1,13 @@
+import * as FullStory from '@fullstory/browser';
+import { Middleware } from 'redux';
+
+const recordAction: Middleware = () => next => action => {
+  if (typeof action !== 'function') {
+    FullStory.log(action.type);
```

Comment:
I wonder if people would want the ability to record certain actions as custom events instead of logs? I guess the main reason not to do this is the limit on the total number of custom events, as well as the rate limiting for the frequency of those custom events?

If these remain logs, should we provide a way to associate different actions as error, info, or warning logs? This could be one way to help make it searchable.

Reply:
Yes, rate limiting is the main driver for not using FS.event given that the limit is fairly conservative (30/min with a burst of 10/sec). The budget per session would likely be exhausted by redux actions alone.

I see the user journey working like this: a frustration event or some other metric guides folks to a particular session. The action log data is an additional resource that sheds light on the issue.

Reply:
> What criteria would we use to distinguish a particular actions as a "warn" vs "error" vs "log"?

I think it'd have to be user configured? While we could try to take the approach of pre-configured actions, I think it might be too specific. Pushing it out to be user configured would allow it to handle almost all situations, and shouldn't be too much work for the end-users.

For example:

```ts
enum LogLevels {
  none,
  debug,
  warning,
  error,
}

const failureRegex = /.*failure.*/g;

const warningLevels = {
  [MyCustomActionType]: true,
}
const logger = recordAction({
  assignLogLevels: (action: Action): LogLevels => {
    if (failureRegex.test(action.type)) return LogLevels.error;
    if (warningLevels[action.type]) return LogLevels.warning;
    return LogLevels.none;
  },
})
```

Reply:
Got it. Would it be all right if we deferred this to the forthcoming PRs around `@fullstory/record-redux-action` and leave as-is for the purposes of this reference app?

Reply:
Absolutely!